test/diff-test.html

```diff
@@ -0,0 +1,128 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>Enhance Custom Element Mixin DOM Diffing Test</title>
+  <link rel="icon" href="data:image/svg+xml,<svg xmlns=%22http://www.w3.org/2000/svg%22 viewBox=%220 0 100 100%22><text y=%22.9em%22 font-size=%2290%22>✨</text></svg>">
+</head>
+<body>
+
+  <my-heading>
+    <button>Submit</button>
+  </my-heading>
+
+  <script type="module">
+    import { runTests } from '@web/test-runner-mocha'
+    import { assert } from '@esm-bundle/chai'
+    import BaseElement from '@enhance/base-element'
+    import TemplateMixin from '@enhance/template-mixin'
+    import CustomElementMixin from '../index.mjs'
+    import morphdom from 'morphdom'
+
+    class CustomElement extends CustomElementMixin(TemplateMixin(BaseElement)) {}
+
+    const MorphdomMixin = (superclass) => class extends superclass {
+      constructor(args) {
+        super(args)
+        this.process = this.process.bind(this)
+      }
+
+      attributeChangedCallback(name, oldValue, newValue) {
+        if (oldValue !== newValue) {
+          this.process()
+        }
+      }
+
+      process() {
+        const tmp = this.render({
+          html: this.html,
+          state: this.state
+        }).trim()
+
+        const updated = document.createElement('div')
+        updated.innerHTML = tmp
+        this.scrubTemplate(updated)
+
+        const root = this.shadowRoot
+          ? this.shadowRoot
+          : this
+
+        const slotEl = document.createElement('div')
+        const slottedElements = root.querySelectorAll('[data-slotted]')
+        for (const value of slottedElements.values()) {
+          slotEl.appendChild(value)
+        }
+        updated.innerHTML = this.expandSlots(slotEl.innerHTML, updated.innerHTML)
+
+        morphdom(
+          root,
+          updated,
+          {
+            childrenOnly: true
+          }
+        )
+      }
+    }
+
+    function MyHeading({ html, state }) {
+      const { attrs = {} } = state
+      const { heading = "default" } = attrs
+      return html`
+        <style>
+          :host > h1{
+            font-size: 2rem;
+            text-transform: uppercase;
+          }
+        </style>
+        <h1>${heading}</h1>
+        <p>${heading}</p>
+        <a href="/foo/${heading}"></a>
+        <slot></slot>
+      `
+    }
+
+    class MyHeadingElement extends MorphdomMixin(CustomElement) {
+      constructor() {
+        super()
+      }
+
+      static get observedAttributes() {
+        return ["heading"]
+      }
+
+      render(args) {
+        return MyHeading(args)
+      }
+    }
+
+    customElements.define("my-heading", MyHeadingElement)
+
+    runTests(()=> {
+      describe('CustomElementMixin Slots', ()=> {
```

Comment:
I need to add another test case for just solo text content not in a wrapping element.